src/Evolution/Systems/NewtonianEuler/Limiters/KxrcfTci.hpp

```diff
@@ -0,0 +1,201 @@
+/// \cite Krivodonova2004.
+///
+/// In summary, this TCI uses the size of discontinuities between neighboring DG
+/// elements to determine the smoothness of the solution. This works because the
+/// discontinuities converge rapidly for smooth solutions, therefore a large
+/// discontinuity suggests a lack of smoothness and the need to apply a limiter.
+///
+/// This implementation
+/// - does not support h- or p-refinement; this is checked by assertion.
+/// - chooses not to check external boundaries, because this adds complexity.
+///   However, by not checking external boundaries, the implementation may not
+///   be robust for problems that feed in shocks through boundary conditions.
+template <size_t VolumeDim, typename PackagedData>
+bool kxrcf_indicator(
+    const double kxrcf_constant, const Scalar<DataVector>& cons_mass_density,
+    const tnsr::I<DataVector, VolumeDim>& cons_momentum_density,
+    const Scalar<DataVector>& cons_energy_density, const Mesh<VolumeDim>& mesh,
+    const Element<VolumeDim>& element,
+    const std::array<double, VolumeDim>& element_size,
+    const Scalar<DataVector>& det_logical_to_inertial_jacobian,
+    const std::unordered_map<Direction<VolumeDim>,
+                             tnsr::i<DataVector, VolumeDim>>&
+        unnormalized_normals,
+    const std::unordered_map<
+        std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>, PackagedData,
+        boost::hash<std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>>>&
+        neighbor_data) noexcept {
+  // Enforce restrictions on h-refinement, p-refinement
+  if (UNLIKELY(alg::any_of(element.neighbors(),
+                           [](const auto& direction_neighbors) noexcept {
+                             return direction_neighbors.second.size() != 1;
+                           }))) {
+    ERROR("The Kxrcf TCI does not yet support h-refinement");
+    // Removing this limitation will require adapting the surface integrals to
+    // correctly acount for,
+    // - multiple (smaller) neighbors contributing to the integral
+    // - only a portion of a (larger) neighbor contributing to the integral
+  }
+  alg::for_each(neighbor_data, [&mesh](const auto& neighbor_and_data) noexcept {
+    if (UNLIKELY(neighbor_and_data.second.mesh != mesh)) {
+      ERROR("The Kxrcf TCI does not yet support p-refinement");
+      // Removing this limitation will require generalizing the surface
+      // integrals to make sure the meshes are consistent.
+    }
+  });
+
+  bool inflow_boundaries_present = false;
+  double inflow_area = 0.;
+  double inflow_delta_density = 0.;
+  double inflow_delta_energy = 0.;
+
+  for (const auto& neighbor_and_data : neighbor_data) {
+    const auto& neighbor = neighbor_and_data.first;
+    const auto& dir = neighbor.first;
+
+    // Skip computations on boundary if the boundary is external
+    // Note: may need to revisit this choice for problems that feed in shocks
```

Comment:
might be worth adding that this would mean that in general the limiter would need to be aware of the boundary condition, which is painful

Reply:
👍 I will expand this note